### PR TITLE
Add support for Couchbase relations

### LINKED ISSUE #9342: @ManyToMany relationships add/remove methods build is broken
##### **Overview of the issue**

add/remove methods for `@ManyToMany` relations doesn't compile since the list of the other entity is missing.

##### **Motivation for or Use Case**

`@ManyToMany`relationships build is broken.

##### **Related issues**

https://github.com/jhipster/generator-jhipster/issues/8991

##### **JHipster Version(s)**

`master` branch

```
jhipster@0.0.0 /Users/tayebchlyah/Project/jhipster/tests/jhipster-sql-relationship
└── generator-jhipster@5.8.1  -> /Users/tayebchlyah/Project/jhipster/projects/generator-jhipster

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp"
    },
    "jhipsterVersion": "5.8.1",
    "applicationType": "monolith",
    "baseName": "jhipster",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "session",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Memory",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "rememberMeKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "clientFramework": "angularX",
    "useSass": false,
    "clientPackageManager": "npm",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": false
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Customer {
  username String required minlength(3) maxlength(250),
  password String required minlength(3) maxlength(250),
  email String required minlength(3) maxlength(250),
  firstName String required minlength(3) maxlength(250),
  lastName String required minlength(3) maxlength(250)
}
entity CustomerOrder {
  date Instant,
  status OrderStatus
}
entity OrderedItem {
  catalogItemId Long,
  name String required minlength(3) maxlength(250),
  price BigDecimal required,
  quantity Integer
}
entity PaymentDetails {
  creditCardNumber String
}
entity ShippingDetails {
  address String
}

enum OrderStatus {
  CONFIRMED,
  SHIPPED,
  DELIVERED
}

relationship OneToOne {
  CustomerOrder{paymentDetails(creditCardNumber)} to PaymentDetails
}
relationship ManyToOne {
  CustomerOrder{customer(username)} to Customer,
  CustomerOrder{shippingDetails(address)} to ShippingDetails
}
relationship ManyToMany {
  CustomerOrder{orderedItems(name)} to OrderedItem
}

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

git version 2.17.2 (Apple Git-113)

node: v11.10.0

npm: 6.7.0

yeoman: 1.8.5

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

##### **Browsers and Operating System**

-   [X] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by tchlyah**
Fix #9342 

This PR needs https://github.com/jhipster/jhipster-core/pull/309, and also [spring-data-couchbase:2.1.6](https://jira.spring.io/issues/?jql=project+%3D+DATACOUCH+AND+fixVersion+%3D+%223.1.6+%28Lovelace+SR6%29%22) release, since it fixes a bug with N1qlJoin (https://github.com/spring-projects/spring-data-couchbase/pull/177).

Depends on https://github.com/jhipster/generator-jhipster/pull/9526

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by MathieuAA** [comment]
I'll review the PR today, but the tests don't pass on JCore. I'll see what I can do about it

**Comment by hdurix** [comment]
Awesome! We could add some relationships between entities in https://github.com/jhipster/jhipster-sample-app-couchbase once it's merged!

File: generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs [hunk 1]
```diff
@@ -23,6 +23,7 @@ let importJsonIgnoreProperties = false;
 let importSet = false;
 let hasDto = dto === 'mapstruct';
 let isUsingMapsId = false;
+let hasRelationship = relationships.length != 0;
```
Comment (MathieuAA) [inline generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs]: use !== instead, for strict equality

**Comment by MathieuAA** [comment]
LGTM!

**Comment by DanielFran** [comment]
@tchlyah Spring-boot 2.1.4 is expected at the end of this month and should include a new release of Spring-data Lovelace sr6. I believe it is better to wait since we expect to launch v6 soon.
What do you think?

**Comment by tchlyah** [comment]
It’s up to you, but I think since we already merged [jhipster-core#309](
https://github.com/jhipster/jhipster-core/pull/309), which activates Couchbase relations, we should do the same for this PR. We can remove related tests waiting for spring-boot release.

Otherwise, I don’t have a problem with that, I just hope that there will be no conflicts until then.

**Comment by MathieuAA** [comment]
The timing isn't great. Maybe the merged PR should be rollbacked until this is merged. That, or we merge this.

**Comment by DanielFran** [comment]
@tchlyah We are expecting spring-boot 2.1.4 today.
PR is ready for it, and as soon as I will merge it, we might then test this PR if @jdubois release the jhipster-core version and upgrade the generator accordingly.

**Comment by DanielFran** [comment]
Closing/Reopenning to launch tests

**Comment by DanielFran** [comment]
@jdubois I am waiting that test finish to merge it after I merge the update to spring-boot 2.1.4.

**Comment by DanielFran** [comment]
@tchlyah We have some tests failing, can you chek please?

**Comment by jdubois** [comment]
Not your fault, we are fixing it!

**Comment by DanielFran** [comment]
Closing/Reopenning to launch tests

**Comment by DanielFran** [comment]
Closing/Reopenning to launch tests

**Comment by DanielFran** [comment]
@tchlyah It seems that there is some issues, see https://travis-ci.org/jhipster/generator-jhipster/jobs/515928525#L5584.
Can you check please?

**Comment by DanielFran** [comment]
Closing/Reopenning to launch tests again

**Comment by DanielFran** [comment]
Closing/Reopenning to launch tests again

**Comment by DanielFran** [comment]
Closing/Reopenning to launch tests again

**Comment by DanielFran** [comment]
@tchlyah Can you please resolve the conflicts?

**Comment by mraible** [comment]
Bump @tchlyah. This PR still has conflicts.

**Comment by DanielFran** [comment]
@tchlyah there is still issues. Can you review it?

**Comment by pascalgrimaud** [comment]
Very good job @tchlyah !
I'm merging this, as the builds passed and as it is opened for long time

**Pull request merged**